### PR TITLE
[Issue #6843] Close open file descriptors

### PR DESCRIPTION
**Pull request by burner761**
I have made sure that when a file is opened using
file = open() 
it is closed later in the file and when is possible converted it to from
with open() as file:

### CODE REVIEW

**Review by afabiani — APPROVED** [review]

**Comment by afabiani** [comment]
@burner761 can you please remove the "worker.state" bin file from this PR?

**Review by afabiani — CHANGES_REQUESTED** [review]
Please remove `worker.state` bin from the PR

File: geonode/tests/utils.py [hunk 1]
```diff
@@ -227,6 +227,19 @@ def upload_file(self, _file):
             data=params,
             ajax=True,
             force_login=True)
+
+        #closes the files
+        if ext.lower() == '.shp':
+            for spatial_file in spatial_files:
+                ext, _ = spatial_file.split('_')
```
Comment (mtnorthcott) [inline geonode/tests/utils.py:234]: Seems unnecessary to compute the filename if `params[spatial_file]` and `params['tif_file']` can be checked to see if they are open file descriptors. Something along the lines of
```python
for spatial_file in spatial_files:
    if isinstance(params[spatial_file], file):
        params[spatial_files].close()

if isinstance(params['tif_file'], file):
    params['tif_file'].close()

**Review by mtnorthcott — CHANGES_REQUESTED** [review]
Minor changes here. Also please check your indentation.

File: geonode/geoserver/helpers.py [hunk 2]
```diff
@@ -580,7 +585,10 @@ def gs_slurp(
        the result of the operation and the errors and traceback if it failed.
     """
     if console is None:
-        console = open(os.devnull, 'w')
+        
+
```
Comment (mtnorthcott) [inline geonode/geoserver/helpers.py:589]: Please remove these extra lines

**Review by mtnorthcott — CHANGES_REQUESTED** [review]

**Pull request merged**